pkg/kubelet/pluginmanager/pluginwatcher/plugin_watcher.go

```diff
@@ -159,7 +159,7 @@ func (w *Watcher) traversePluginDir(dir string) error {
 func (w *Watcher) handleCreateEvent(event fsnotify.Event) error {
 	klog.V(6).Infof("Handling create event: %v", event)
 
-	fi, err := os.Stat(event.Name)
+	fi, err := os.Lstat(event.Name)
```

Comment:
This doesn't look right to me. If my plugin set up its socket in /myplugin/mysocket.sock, then linked /plugindir/myplugin -> /myplugin, this change means the symlink wouldn't get traversed, right?

/hold

Reply:
I think with or without this change, the behavior of the code will not change. For linux, it uses os.Stat() twice

1. The first os.Stat() is to check whether path is the dir or not: (The assumption here is if the path is a dir, it should not be a socket file)

Before fix, if path is a symlink, and target path is a dir, then the code will return immediately 
After changing to os.Lstat(),  if path is a symlink, it will continue to call IsUnixDomainSocket in side of this function it will call os.Stat() again in step 2

2. The second os.Stat()
This check will determine whether file is a socket file or not. We are not changing this os.Stat() here for Linux.
Windows uses a different way to check domain socket file, since file Mode returned by os.Stat() does not work for Windows.

So the first os.Stat (now it changed to os.Lstat()) should not affect the final result. Actually the first check is not very necessary in my option since the actual check of domain socket is in the second check.

Reply:
> Before fix, if path is a symlink, and target path is a dir, then the code will return immediately

if path was a symlink, and target path was a dir, linux would go to `return w.traversePluginDir(event.Name)`, right? now it will hit the `klog.V(5).Infof("Ignoring non socket file %s", fi.Name())` path, right?

Reply:
right, I missed w.traversePluginDir(event.Name) part.

Thanks a lot!